### ISSUE TITLE
arangodump corrupts unloaded collections

### ISSUE THREAD

**Issue by zpolisensky** [arangodb/arangodb#4626]
## my environment running ArangoDB
I'm using the latest ArangoDB of the respective release series:
- [X] 3.3

Mode:
- [X] Single-Server

Storage-Engine:
- [X] mmfiles

On this operating system:
- [X] Linux 
  - [X] Debian .deb

Problem description:
Command arangodump corrupts collections which are in "unloaded" state when command is issued. 

These are the steps to reproduce:
1) open linux terminal (ssh to server)

2) run command "arangodump --output-directory /tmp/arangodump --server.database somedb --server.username someuser --server.password somepass"

3) dump crashes on the first "unloaded" collection with error "2018-02-20T09:20:28Z [4365] ERROR got error from server: HTTP 404 (Not Found): ArangoError 1203: collection not found"

4) affected collection changes its state from "unloaded" to "corrupted"

Version:
3.3.3

architecture: 64bit
arm: false
asan: false
asm-crc32: true
assertions: false
boost-version: 1.62.0
build-date: 2018-01-25 20:21:22
build-repository: tags/v3.3.3-0-g5819c24e24
compiler: gcc [5.4.0]
cplusplus: 201103
debug: false
endianness: little
failure-tests: false
fd-client-event-handler: poll
fd-setsize: 1024
full-version-string: ArangoDB 3.3.3 [linux] 64bit, using jemalloc, VPack 0.1.30, RocksDB 5.6.0, ICU 58.1, V8 5.7.492.77, OpenSSL 1.0.1t  3 May 2016
icu-version: 58.1
jemalloc: true
maintainer-mode: false
openssl-version: OpenSSL 1.0.1t  3 May 2016
platform: linux
reactor-type: epoll
rocksdb-version: 5.6.0
server-version: 3.3.3
sizeof int: 4
sizeof void*: 8
sse42: false
unaligned-access: true
v8-version: 5.7.492.77
vpack-version: 0.1.30
zlib-version: 1.2.11

**Comment by maxkernbach** [comment]
Hi @zpolisensky,

i tried to reproduce it with the current version 3.3.3 using Ubuntu 17.04 and a custom user/database but arangodump worked for me. Collections, that are unloaded before executing arangodump, are loaded when it is finished. Is there anything else to consider? Does this error only occur when using a non-root user and not the _system database?

**Comment by zpolisensky** [comment]
Hi @maxkernbach ,

I did more testing and found out that it does not happen if Arango was upgraded from any previous version so it must be clean installation (it fine just to purge it from the system: sudo apt purge arangodb3). My OS is Debian 8.0. I'm using non-root, read-only user on not the _system DB.
![dump-error](https://user-images.githubusercontent.com/36234956/36422009-eb4ff95e-163a-11e8-84bb-f0f30d767dea.png)

**Comment by maxkernbach** [comment]
Could you please provide further information on your collection size and count? This would be helpful to reproduce your issue. Thanks!

**Comment by zpolisensky** [comment]
Hi Max,
here is more information:
1. dump works fine if authentication is disabled "authentication = false"
2. dump work fine if user used for it has "Administrative" access
3. dump fails if user has only "Access" level of permissions
So it looks that user with "Access" permissions do not have ability to load collection as it works earlier versions of Arango.
Good news are that collection switch to corrupted state are OK after Arango restart.

**Comment by zpolisensky** [comment]
OK, I found how to make it work for every user:
Everything is done in User menu on user configuration level:
![2018-03-01 13_54_20-arangodb web interface](https://user-images.githubusercontent.com/36234956/36845866-a1ec8ed8-1d58-11e8-8021-ebf2adca019a.png)

1. Add user "Access" level privileges on database
2. Click on database name to drop down to collections level
3. Do down to "*" (Default access level for collections) and set also "Access" level
4. Dumb now works

**Comment by dothebart** [comment]
Hi, 
I'm trying to reproduce this but failed. 
I've used collections in the `_system` database and the `test` database; 

Cutting two screenhots from the permissions of the user to one:
![permissions](https://user-images.githubusercontent.com/5013077/36846415-ca4fe1a2-1d5a-11e8-8fb1-40f18e67c4a7.png)

The user needs access to the `test` database, else dumping fails from the start:
```
2018-03-01T13:09:04Z [3445] ERROR Could not connect to endpoint 'tcp://127.0.0.1:8529', database: 'test', username: 'test'
2018-03-01T13:09:04Z [3445] FATAL Error message: 'got error from server: HTTP 401 (Unauthorized)'
```
can you pin in my screenshot which permission I should change so this happenes?

**Comment by dothebart** [comment]
ah, hang on - just a second to early - got it reproduced.

**Comment by dothebart** [comment]
will be addressed with https://github.com/arangodb/arangodb/pull/4725

**Comment by OmarAyo** [comment]
Hi @zpolisensky ,

Thanks for reporting this bug

We have released 3.3.4, and it includes a fix for it
I am closing this ticket now. Please feel free to reopen if the problem persists
